src/Shared/UnitTests/ObjectModelHelpers.cs

```diff
@@ -1581,6 +1587,40 @@ private static TempPaths GetTempPaths()
             }
         }
 
+        internal class TemporaryEnvironment : IDisposable
+        {
+            private bool _disposed;
+            private readonly string _name;
+            private readonly string _originalValue;
+
+            public TemporaryEnvironment(string name, string value)
```

Comment:
Would it make sense for this to take an `IDictionary<string, string>` so you could have one or more variables set?